stan/math/prim/fun/matrix_exp_action_handler.hpp

```diff
@@ -54,39 +90,66 @@ class matrix_exp_action_handler {
       A(i, i) -= mu;
     }
 
-    int m{0}, s{0};
-    set_approximation_parameter(A, t, m, s);
-
-    Eigen::MatrixXd res(A.rows(), b.cols());
-
-    for (int col = 0; col < b.cols(); ++col) {
-      bool conv = false;
-      Eigen::VectorXd B = b.col(col);
-      Eigen::VectorXd F = B;
-      const auto eta = std::exp(t * mu / s);
-      for (int i = 1; i < s + 1; ++i) {
-        auto c1 = B.template lpNorm<Eigen::Infinity>();
-        if (m > 0) {
-          for (int j = 1; j < m + 1; ++j) {
-            B = t * A * B / (s * j);
-            auto c2 = B.template lpNorm<Eigen::Infinity>();
-            F += B;
-            if (c1 + c2 < tol * F.template lpNorm<Eigen::Infinity>()) {
-              conv = true;
-              break;
-            }
-            c1 = c2;
-          }
-        }
-        F *= eta;
-        B = F;
-        if (conv) {
+    int m, s;
+    set_approx_order(A, b, t, m, s);
+
+    double eta = exp(t * mu / s);
+
+    const auto& b_eval = b.eval();
+    Eigen::MatrixXd f = b_eval;
+    Eigen::MatrixXd bi = b_eval;
+
+#define MAT_OP_INF_NORM(X) X.cwiseAbs().rowwise().sum().maxCoeff()
```

Comment:
Not a huge fan of this #define and #undef approach. Given that its used only 3 times, can we just use it everywhere?

Reply:
replaced with function